### ISSUE TITLE
ImageToVieportZoom error

### ISSUE THREAD

**Issue by btzr-io** [openseadragon/openseadragon#1737]
Getting this error when having  more that one tile on the viewer;
```
[Viewport.imageToViewportZoom] is not accurate with multi-image.
```

What should I use instead for `multi-image` ?

**Comment by iangilman** [comment]
Each image potentially has a different zoom, so the viewport version of this function is kind of meaningless with multi-image. That's why it's an error, and why there's no way to suppress it. To get the viewport zoom for an individual image, use `TiledImage.imageToViewportZoom()`. If you want to get all of the zooms for the loaded images, you can do something like:

```
var count = viewer.world.getItemCount();
var imageZoom = 2; // For example

for (var i = 0; i < count; i++) {
  console.log(viewer.world.getItemAt(i).imageToViewportZoom(imageZoom));
}
```

I guess the error could be improved by mentioning the TiledImage version.

**Comment by btzr-io** [comment]
Is it possible to get single zoom for all tiles ?

**Comment by btzr-io** [comment]
I'm trying to set the `maxZoom` using `viewport.imageToVieportZoom()` so the you can only zoom in until the image is displayed on the original / real size.

It seems to work fine for this use case :confused:

**Comment by btzr-io** [comment]
Find a workaround for now, just get the largest image zoom:
```js
  getTargetZoom = (scale = 1) => {
    let zooms = []
    const { viewport, world } = this.viewer
    const count = world.getItemCount()

    for (let i = 0; i < count; i++) {
      zooms[i] = world.getItemAt(i).imageToViewportZoom(scale)
    }

    return Math.max(zooms)
  }
```

**Comment by iangilman** [comment]
Yup, that's a perfect solution. I wouldn't call it a workaround... it's exactly what you're supposed to do in order to get an accurate result. :)

**Comment by Ekjot07** [comment]
Is this issue still open to be worked upon? if so I would like to give it a shot

**Comment by iangilman** [comment]
@Ekjot07 It is open! It would be great if you could fix it up. Do you understand the issue? It's just a little bit of documentation that's needed.

**Comment by Pritam1729** [comment]
Do you want alternate name for "Multi-image" or do you want to resolve the error "[Viewport.imageToViewportZoom]"?

**Comment by iangilman** [comment]
@Pritam1729 Are you looking to contribute a fix for this issue? If so, please do!

All we need is a little update to the error message. Something like: 

```
[Viewport.imageToViewportZoom] is not accurate when there are multiple images in the viewer. Instead, use [TiledImage.imageToViewportZoom] on the specific image you are interested in.
```

**Comment by jessemchung** [comment]
@iangilman Hello, if it is still free, I would like to work on this issue

**Comment by msalsbery** [comment]
> @iangilman Hello, if it is still free, I would like to work on this issue

Go for it, and thank you!

**Comment by SuperexMack** [comment]
can i work on this issue ? or is it fixed?

**Comment by msalsbery** [comment]
@jessemchung are you working on this?

**Comment by jessemchung** [comment]
Hey sorry, yeah I have not gotten anywhere yet with this.  Feel free to
work on it

**Comment by msalsbery** [comment]
@SuperexMack there you go, thank you! 👆

**Comment by wmavis** [comment]
It looks like the error message and documentation was updated in the branch listed above.  Should this be closed or is there more to do?

**Comment by iangilman** [comment]
@wmavis Good catch! You're absolutely right. Thank you.
